actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -481,6 +487,8 @@ def []=(name, options)
           options = { value: options }
         end
 
+        options[:cookie_name] = name
```

Comment:
Won't this break old Rails apps as we're always writing the `cookie_name` to the cookie? E.g. would a cookie from 5.2 (e.g. no metadata) be readable on 6.0? And would a metadata attached cookie from 6.0 be readable on 5.2? This is one of the most important things we have to keep in mind, so do our tests test these scenarios sufficiently and do they do so in a sufficiently realistic manner?

Reply:
I still don't like assigning straight to the options here. They'll be committed in the cookie body after all. Have you tried passing the name in to commit or moving the metadata construction up to here?